src/platform/K32W/KeyValueStoreManagerImpl.cpp

```diff
@@ -24,76 +24,189 @@
 #include <platform/internal/CHIPDeviceLayerInternal.h>
 
 #include <lib/support/CHIPMem.h>
+#include <list>
 #include <platform/K32W/K32WConfig.h>
 #include <platform/KeyValueStoreManager.h>
 #include <string>
 
 #include "PDM.h"
 
+#include <unordered_map>
+
 namespace chip {
 namespace DeviceLayer {
 namespace PersistedStorage {
 
 /* TODO: adjust this value */
-#define MAX_NO_OF_KEYS 255
+#define MAX_NO_OF_KEYS 20
 
 KeyValueStoreManagerImpl KeyValueStoreManagerImpl::sInstance;
 
-CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t value_size, size_t * read_bytes_size,
-                                          size_t offset_bytes)
+/* hashmap having:
+ * 	- the matter key string as key;
+ * 	- internal PDM identifier as value;
+ */
+std::unordered_map<std::string, uint8_t> g_kvs_map;
+
+/* list containing used PDM identifiers */
+std::list<uint8_t> g_key_ids_list;
+
+/* max no of bytes for a key */
+#define MAX_KEY_VALUE          255
```

Comment:
same: use C++ constants. Ideally group constants in one place.

Use the name of `kMaxKeyValueBytes` then the comment is not needed.

Reply:
Done, used constexpr and grouped constants in one place.